go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -512,15 +522,20 @@ func (c *Controller) RequestFullSync(nodeInfos []nodeInfo) {
 	// Resync node infos and node IP templates.
 	c.syncNodeInfos(nodeInfos)
 
-	// Resync services.
-	services, err := c.serviceLister.List(labels.Everything())
-	if err != nil {
-		klog.Errorf("Cached lister failed!? %v", err)
-		return
-	}
+	// Resync all services unless we're processing the initial node tracker sync (in which case
+	// the service add will happen at the next step in the controller services Run())
+	c.nodeTrackerInitialSyncDoneLock.RLock()
+	defer c.nodeTrackerInitialSyncDoneLock.RUnlock()
+	if c.nodeTrackerInitialSyncDone {
+		services, err := c.serviceLister.List(labels.Everything())
+		if err != nil {
+			klog.Errorf("Cached lister failed!? %v", err)
```

Comment:
shouldn't this be .... return the error and fail hard/retry? or do we want to keep it as is..

Reply:
You're right, I simply kept the existing logic. If I look at how node tracker sync works, it has no retry mechanism at all o_O
It might be something to explore for our tech debt, maybe?

Reply:
yeah sounds good, deff don't want you to go out of scope, perhaps open a JIRA tracker for this please! thanks

Reply:
Here it is: https://issues.redhat.com/browse/SDN-4483